python/ray/autoscaler/v2/instance_manager/storage.py

```diff
@@ -0,0 +1,173 @@
+import copy
+from abc import ABCMeta, abstractmethod
+from collections import defaultdict
+from threading import Lock
+from typing import Dict, List, Optional, Tuple
+
+
+class Storage(metaclass=ABCMeta):
+    """Interface for a storage backend that stores the state of nodes in the cluster.
+
+    The storage is versioned, which means that each successful stage change to the
+    storage will bump the version number. The version number can be used to
+    implement optimistic concurrency control.
+
+    Each entry in the storage table is also versioned. The version number of an entry
+    is the last version number when the entry is updated.
+    """
+
+    @abstractmethod
+    def batch_update(
+        self,
+        table: str,
+        mutation: Dict[str, str] = {},
```

Comment:
```suggestion
        mutation: Dict[str, str] = None,
```
Giving the default {} or [] is an anti pattern in python https://dollardhingra.com/blog/python-mutable-default-arguments/.

Reply:
wow..

Reply:
I know right lol... it is a weird language